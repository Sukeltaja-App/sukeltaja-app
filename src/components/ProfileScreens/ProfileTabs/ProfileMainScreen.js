import React from 'react'
import { connect } from 'react-redux'
import { View } from 'react-native'
import { Avatar, Text, Button } from 'react-native-elements'

import { logout } from '../../../store'
import { getServerListener } from '../../../ServerListener'
import userService from '../../../services/users'
import styles from '../../../styles/global'
import colors from '../../../styles/colors'

export const ProfileMainScreen = (props) => {
  const uri = 'https://upload.wikimedia.org/wikipedia/commons/5/50/USS_Scorpion_sail.jpg'

  const navigate = (value) => props.navigation.navigate(value)

  const logoutButton = () => {
    const { logout } = props

    const serverListener = getServerListener()

    serverListener.disconnect()

    logout()
    userService.setToken(null)

    navigate('LoginScreen')
  }

  return (
    <View style={styles.centered}>
      <Avatar source={{ uri }} size="xlarge" rounded />
      <Text h2>{props.username}</Text>
      <View style={styles.bottom}>
        <Button
          onPress={() => logoutButton()}
          buttonStyle={{ backgroundColor: colors.red }}
          title="Kirjaudu ulos"
          raised
        />
      </View>
    </View>
  )
}

const mapStateToProps = (state) => ({ username: state.user.username })

export default connect(
  mapStateToProps,
  { logout }
)(ProfileMainScreen)
