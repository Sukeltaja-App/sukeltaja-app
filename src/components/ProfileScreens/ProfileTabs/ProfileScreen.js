import React from 'react'
import { connect } from 'react-redux'
import { View } from 'react-native'
import { Image, Text, Button } from 'react-native-elements'

import { logout } from '../../../store'
import { getServerListener } from '../../../ServerListener'
import userService from '../../../services/users'
import styles from '../../../styles/global'
import colors from '../../../styles/colors'

export const ProfileMainScreen = (props) => {
  const uri = require('../../../pictures/mobiililogot_kolmio.png')

  const navigate = (value) => props.navigation.navigate(value)

  const logoutButton = () => {
    const { logout } = props

    const serverListener = getServerListener()

    serverListener.disconnect()

    logout()
    userService.setToken(null)

    navigate('LoginScreen')
  }

  return (
    <View style={styles.centered}>
      <Image source={ uri } />
      <Text h2>{props.username}</Text>
      <View style={styles.bottom}>
        <Button
          onPress={() => logoutButton()}
          buttonStyle={{ backgroundColor: colors.red }}
          title="Kirjaudu ulos"
          raised
        />
      </View>
    </View>
  )
}

const mapStateToProps = (state) => ({ username: state.user.username })

export default connect(
  mapStateToProps,
  { logout }
)(ProfileMainScreen)
