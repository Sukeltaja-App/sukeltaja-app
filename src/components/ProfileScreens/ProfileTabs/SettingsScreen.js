import React from 'react'
import { View } from 'react-native'

import styles from '../../../styles/global'

const SettingsScreen = () => <View style={styles.noPadding} />

export default SettingsScreen
