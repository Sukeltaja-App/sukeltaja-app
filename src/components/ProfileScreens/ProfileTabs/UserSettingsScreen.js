import React from 'react'
import { View } from 'react-native'

const UserSettingsScreen = () => <View />

export default UserSettingsScreen
