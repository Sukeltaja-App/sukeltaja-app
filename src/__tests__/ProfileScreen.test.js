import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { ProfileScreen } from '../components/ProfileScreens/ProfileTabs/ProfileScreen'

let wrapper

beforeAll(() => {
  wrapper = shallow(<ProfileScreen />)
})

describe('<ProfileScreen/>', () => {
  it('renders the profile screen snapshot', () => {
    const profileScreen = renderer
      .create(wrapper)
      .toJSON()

    expect(profileScreen).toMatchSnapshot()
  })
})
