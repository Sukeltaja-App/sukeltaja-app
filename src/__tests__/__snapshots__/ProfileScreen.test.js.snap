// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileScreen/> renders the profile screen snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "flex": 1,
    }
  }
/>
`;
